codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/RestBuilderGeneratorBase.java

```diff
@@ -286,11 +286,6 @@ protected void renderUriBuilder(OperationShape operation, Shape inputShape) {
                                     + getter + ")"
                     );
                 }
-                writer
-                        .openBlock("if $1L.empty?", getter)
```

Comment:
If someone has disabled validation, will we still skip validating required?  If so - would that cause this code to fail here with an unexpected error? (I think it may just result in an incorrect url being produced.  Either way, we *may* still want to keep these checks since we do allow configuration to disable validation).

Reply:
Yes it would skip validating required. It would fail with an unexpected error (bad URL). That's the behavior today in V3 - `Aws::S3::Client.new(validate_params: false).abort_multipart_upload(key: 'foo')` will be `DELETE /foo` (missing uploadId). If we were guarding against everything in the URI, we'd want to check query params too. Personally I lean towards doing all or nothing with required and a way to configure that off.

Reply:
I think the S3 case is a bit specific - as far as we know that is the only service where the presence of user input query parameters changes which operation is used so I dont think validating query param is the same as validating host label presence.    I'd personally lean towards keeping this check and not allowing any configuration that opts out of this.

But I'll defer to you on it if you want to remove it.